### DISCUSSION TITLE
A "stealth mode" for submitting posts that admins don't want to appear on the main feed

### DISCUSSION THREAD

**Discussion by ellativity** [forem/forem#15772] [Feature Requests]
We love using Forems as knowledge bases for communities, as well as spaces for active and heartfelt discussions.

One of the issues we've noticed with adding knowledge-type information to Forems is that they can heavily skew the vibe of the feed, especially for members who are following relevant tags. This is why I'm requesting adding a feature that allows admins to keep certain posts off the feed. 

Forem Admins should have more control over what content gets displayed on the feed for their Forem. While we have basic moderator tools to influence the **position** of content on the feed, we don't currently have a way to remove content from the feed without removing it from the site altogether (i.e. unpublish, or delete).

Some suggestions are
- a "remove from feed" checkbox on either `/moderate` or `/manage` (appending one of these to the end of a post URL navigates to specific Admin-only or Admin-and-author options pages) allowing Admins the ability to remove any item from the feed that appears on it (the ideal choice, with caveats described below)
- a "hide from feed" option in the Post Editor allowing authors to specify that they want to hide the post from the feed (this should be toggled by Forem Admins, so that it's an option rather than default - see below for more context)
- a "hide from main feed" checkbox for the Tag Admin page (Admin-only page accessible from any tag page by appending `/admin`) that automatically excludes any content tagged with the tag

The ability to hide _specific posts_ will be beneficial for Forem Admins wanting to add searchable content to their Forem without cluttering up the main feed, such as "how to" guides for product Forems, or legacy company documentation for internal team Forems. 

The ability to hide _entire tag feeds_ will be beneficial for Forems with several specific user groups, such as brand advocates vs support seekers, allowing for certain information to be compartmentalized on publication with a specific tag (I'd recommend that this is set up so that certain preconditions would need to be fulfilled elsewhere in the tag admin, such as checking the "supported" box - "hide from main feed" could even be a conditional box that is only checkable if "supported" is checked first - and specifying a Tag Moderator - see previous paragraph for reasons why extra vigilance is required for this type of functionality). 

#### Worth noting
**Since hidden content will be less visible and less likely to be reported to Admins if it breaks site rules, I recommend setting up any option to hide content _so that only Admins can use it by default_ (i.e. using only `/moderate` and not `/manage` options, unless it can be toggled on/off by the Admin for all users), to prevent abuse of the tool by non-Admins.**

### Additional context
I don't see this solution as an alternative to allowing Forem Admins greater control over the text on all pages of their Forems or [custom user groups](https://github.com/forem/rfcs/issues/337), but a separate solution to a related issue. In addition, Forem Admins having the ability to assign user groups could complement this feature nicely, as specific user groups could be given permission to stealth post.

I see this feature serving Forem-hosted Forems like forem.dev (knowledge base) and internal Forems (knowledge base), as well as specific partner Forems (knowledge base, brand advocates, special interest groups). I consider this to be a sooner-rather-than-later feature that will meet a lot of community needs and improve both Admin and user experiences.

**Comment by FounderCasey** [comment]
I absolutely LOVE this suggestion. There are countless times where, as an admin, I want to post something quietly and not take over the feed.

The "remove from feed" suggestion seems like a great way to get started

**Comment by michael-tharrington** [comment]
I love all three of these suggestions:

> 1. a "remove from feed" checkbox on either /moderate or /manage (appending one of these to the end of a post URL navigates to specific Admin-only or Admin-and-author options pages) allowing Admins the ability to remove any item from the feed that appears on it.

This is really cool for admin to be able to remove posts from the feed manually. Sometimes a user will write post called "Test" simply to test out the site. This doesn't generally feel worthy of removal and the author typically doesn't need intervention (unless they make a habit of posting lots of posts like this). It would be nice for admin to just be able to click a remove from feed button.

> 2. a "hide from feed" option in the Post Editor allowing authors to specify that they want to hide the post from the feed

Also great. Sometimes a user wants to post notes for themselves without broadcasting to the rest of the site. Sometimes a new user joins and imports their articles via RSS and wants to post them all at once. Sometimes site admin want to post a bunch of resources specific to the site, but they don't want to make a big splash.

> 3. a "hide from main feed" checkbox for the Tag Admin page (Admin-only page accessible from any tag page by appending /admin) that automatically excludes any content tagged with the tag

Love this! I see it as basically the opposite of supported, haha. Admin can't block tags from being created, but it'd be great if admin could do something to hide content from tags that they don't want the community to focus on. There's a related discussion [here on forem.dev](https://forem.dev/ildi/is-it-possible-to-delete-hide-tags-16km) where a Forem Creator expresses the desire to hide or delete tags.

**Reply by jeremyf** [reply]
Question about "hide from main feed"?

Let's say I have a post that I tag with three tags.  Two of those tags we've indicated are "hide from main feed".  What do I do with the post?  Should I show it or hide it in the feed?  What if two of the tags are not hidden but one is?

**Reply by JennieOcken** [reply]
If a post uses a tag that is "hidden" from the main feed, the post is hidden regardless of what other tags have been applied.  In other words, this is an _override_ of the post because it has been "hidden".

**Reply by jeremyf** [reply]
Next question regarding state change.  

```gherkin
Given Tag A has "show posts in feed" set to `false`
When an Admin changes Tag A's "show posts in feed" to `true`
Then all previously published Articles with that tag will now be available to show in the feed.
```

My assumption is that yes, once you change a Tag's setting, it propagates.

**Reply by JennieOcken** [reply]
Yes, it propagates.

Now, we Community Success and Creators will need to be aware of these interactions so I expect a troubleshooting situation like this to occur:

Post 1 has tags A, B, C.  A and B are set to hidden
Admin changes A to be shown.
Post 1 still doesn't show because tag B is still set to hidden.

Given that it is a bit hard to tell what is going to show up in the feed, I do imagine questions like "why is post 1 not showing" and we should _start_ after this feature by looking at tag visibility status.

FYI @ellativity and @michael-tharrington

**Comment by leewynne** [comment]
Love it !!!!

**Comment by jeremyf** [comment]
Follow-up question, when we say "Omit from Feed" which all "feeds" do we mean?

Assume that for the following table, the Article's "show posts in feed == `false`".  That is either set explicitly on the Article or is derived from one or more of the Article's tags.  Here's some of the scenarios that I want to consider

| Feed | On Article | Derived from Article's Tag |
|-|-|-|
| Relevant | Do not show | Do not show |
| Latest | ? | ? |
| Top (Week, Month, Year, Infity) | ? | ? |
| Tags > Relevant | Show? † | Show
| Tags > Latest | Show? † | Show |
| Tags > Top | Show? † | Show |

† - For an Article with a tag that is "show posts in feed == `false`", I assume we want to show it in the Tag's various feeds.

**Reply by JennieOcken** [reply]
| Feed                            | On Article  | Derived from Article's Tag |
|---------------------------------|-------------|----------------------------|
| Relevant                        | Do not show | Do not show                |
| Latest                          | Do not show | Do not show                |
| Top (Week, Month, Year, Infity) | Do not show | Do now show                |
| Tags > Relevant                 | Show        | Show                       |
| Tags > Latest                   | Show        | Show                       |
| Tags > Top                      | Show        | Show                       |

I started with the "derived from article's tag" because it was easier.  In that case we have an entire tag that is being used for something like documentation or historical.  So the tag page still exists and the articles need to show on the tag pages otherwise what's the point.  But the articles should not be in any of the main feeds (relevant, latest, top) because the purpose of the "hide" function is to make those posts more "low visibility" because they are documentation or historical.  

For the article specific hide, we are dealing with a situation where that specific article has been identified either because of the nature of it's desired audience or the timeliness of the content.  I agree we show it in the tag pages still.  At that point, I am thinking about consistency of behavior.  I think it wouldn't make sense to have the individual "hide" work radically different from the tag based hide.  So... same settings.

So the logical rule here is "hide from feed" is referring to all the feeds on the main page only.

**Reply by ellativity** [reply]
I agree with @JennieOcken's logic here (maybe because this is exactly what I had in mind: the initial request started out as having individual posts "stealthed", but when I thought about how they would then be accessed/archived then the next step of having entirely "stealthed" tags made sense).

Related: I'd love to find a word that's better than "hidden" because that implies that the post doesn't show up anywhere, but if it's populating tag-specific feeds it's not completely hidden from view.

**Reply by JennieOcken** [reply]
Naming things is hard

Thesaurus.com...

Private, Secret, buried, clandestine, concealed, obscure, undisclosed, cloaked, masked, screened, ...

:getsdistracted: On the QT, private, sneaky, stealthy, under wraps, undercover, shrouded

So, we aren't hiding it from _all_ feeds, just the _main feeds.  So maybe we go more explicit here

`Remove from main feed page`

Then we can add a little ? tag on that which says `Selecting this will remove this post/posts with this tag from all the main page feeds.  It will still appear on tag pages`

**Reply by ellativity** [reply]
That seems clear and helpful!

**Reply by amywtlin** [reply]
I also agree with Jennie's logic here. Also wanted to suggest a name: post in silence. Essentially the posts hidden from the feed are still accessible, just not from the home feed, no matter which sorting (relevance/top/latest) users are on. So it's more about posting an article without interrupting users' flow of consuming articles on the home feed - where I feel "in silence" is a good option.

What do you think @JennieOcken @ellativity ?

It looks like we are leaning towards having the "hide from feed" configuration on a tag level - when a tag's "hide from feed" is turned on, all posts won't show up in home feed. Correct?

**Reply by amywtlin** [reply]
Oh oops. I didn't read through the comment from Jennie and didn't realize we have a solution already. Feel free to omit my comment!

One thing I would add though - I think it's better to have the configurations at the tag level, instead of post level. As we slowly move towards Roles System, where we can configure behaviors of a content group (for example, post upon approval), this would be the same level of tags. This is good for consistency also while migrating to Roles System eventually.

**Comment by thomasbnt** [comment]
Good suggestion 🌱👍🏼

**Comment by michael-tharrington** [comment]
I think this feature would also be really cool for folks who want to edit and repost an article that has already been live on the site. 

Currently, if you edit an older article and repost it, it hits folks notifications, which in most cases isn't ideal. But, it might be nice to give authors the choice if they wanna amplify their newly edited post (allow it to hit folks notifications and the feed) or if they just wanna silently edit (not hit notifications/feed).

**Reply by amywtlin** [reply]
I might be skeptical, but my immediate reaction is how this can be easily gamed to manipulate the feed 😂 But I do get that it could be a good way to promote the edited version of the post, given that the newly edited part deserves it.
